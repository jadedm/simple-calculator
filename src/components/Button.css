.button-wrapper .button{
  flex-basis: 25%;
  height: 3em;
  line-height: 1.4em;
  border: none;
  outline: 1px solid #888;
  background: #e0e1e6;
  color: #888;
  font-weight: bold;
  display: flex;
  justify-content: center;
  align-items: center;
  cursor: pointer;
}

.button-wrapper .operator {
  background-color: #fe9241;
  color: #ffffff;
}

.button.operator:last-child {

    flex-grow: 1;

}