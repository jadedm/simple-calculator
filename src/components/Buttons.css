.button-wrapper {
  display: flex;
  justify-content: stretch;
  align-items: center;
  font-weight: lighter;
  font-size: 1.4em;
  background-color: #e0e1e6;
  color: #888888;
  outline: 1px solid #888888;
  flex-flow: row wrap;
}
