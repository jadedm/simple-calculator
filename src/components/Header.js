import React from 'react';

const Header = () => {
	return(
		<h1 style={{ textAlign: 'center'}}>Simple Calculator</h1>
	);
}

export default Header;