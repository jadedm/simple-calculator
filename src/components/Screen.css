.calc-screen {
  height: 2.5em;
  display: flex;
  justify-content: flex-end;
  align-items: center;
  font-weight: bold;
  font-size: 2em;
  background-color: #000000;
  color: #ffffff;
  padding: 0.1em 0.5em;
  outline: 1px solid #888888;
  border: none;
  overflow: auto;
}

.error { 
  color: tomato;
  text-align: center;
  font-weight: bold;
  text-decoration: underline;
  font-style: italic;
}