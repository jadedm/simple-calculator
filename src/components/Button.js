import React from 'react';

import './Button.css';

const isOperator = val => {
  return !isNaN(val) || val === "." || val === "=";
};

export const Button = ({ buttonEl, onButtonClick }) => {
	return (
		<div 
			className={`button ${isOperator(buttonEl) ? '' : "operator"}`}
			onClick={() => { onButtonClick(buttonEl) }}>
			{buttonEl}
		</div>
	);
};

export default Button;