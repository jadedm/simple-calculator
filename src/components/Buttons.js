import React, {Component} from 'react';
import { connect } from 'react-redux';

import './Buttons.css';
import Button from './Button.js';
import { ButtonElements } from './../utils/ButtonElements';
import { addElement, clearScreen, calculateValue } from './../actions';

export class Buttons extends Component {
  
  handleClick = (el) => {
    switch(el) {
      case 'Clear':
        this.props.clearScreen();
        break;

      case '=':
        this.props.calculateValue(this.props.value)
        break;

      default:
        this.props.addElement(el)
    }
  }

  render(){
    return (
      <div className="button-wrapper">        
        {ButtonElements.map(row => {
          return row.map((button, index) => (
            <Button key={button} buttonEl={button} onButtonClick={this.handleClick}/>
          ))
        })}
      </div>
    );
  }
}

const mapStateToProps = (state) => {
	return { value: state.calculator.value }
}

export default connect(mapStateToProps, { addElement, clearScreen, calculateValue  })(Buttons);