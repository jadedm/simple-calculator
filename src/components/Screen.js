import React from 'react';
import { connect } from 'react-redux';

import './Screen.css';

export const Screen = ({ value, error }) => {
	return (
		<React.Fragment>
			{ error && <p className="error">{ error.toUpperCase() }</p> }
			<div className='calc-screen'>{value}</div>
		</React.Fragment>
	);
};

const mapStateToProps = (state) => {
	return {
		value: state.calculator.value,
		error: state.calculator.error
	}
}

export default connect(mapStateToProps)(Screen);