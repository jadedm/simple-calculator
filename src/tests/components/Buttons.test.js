import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { Buttons } from './../../components/Buttons';
import elements from './../fixtures/buttonElements';

configure({ adapter: new Adapter() });

test('should render individual buttons', () => {
	const wrapper = shallow(<Buttons />);
	expect(toJSON(wrapper)).toMatchSnapshot();
});