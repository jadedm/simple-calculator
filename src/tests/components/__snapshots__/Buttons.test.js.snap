// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render individual buttons 1`] = `
<div
  className="button-wrapper"
>
  <Button
    buttonEl="7"
    key="7"
    onButtonClick={[Function]}
  />
  <Button
    buttonEl="8"
    key="8"
    onButtonClick={[Function]}
  />
  <Button
    buttonEl="9"
    key="9"
    onButtonClick={[Function]}
  />
  <Button
    buttonEl="+"
    key="+"
    onButtonClick={[Function]}
  />
  <Button
    buttonEl="4"
    key="4"
    onButtonClick={[Function]}
  />
  <Button
    buttonEl="5"
    key="5"
    onButtonClick={[Function]}
  />
  <Button
    buttonEl="6"
    key="6"
    onButtonClick={[Function]}
  />
  <Button
    buttonEl="*"
    key="*"
    onButtonClick={[Function]}
  />
  <Button
    buttonEl="1"
    key="1"
    onButtonClick={[Function]}
  />
  <Button
    buttonEl="2"
    key="2"
    onButtonClick={[Function]}
  />
  <Button
    buttonEl="3"
    key="3"
    onButtonClick={[Function]}
  />
  <Button
    buttonEl="/"
    key="/"
    onButtonClick={[Function]}
  />
  <Button
    buttonEl="."
    key="."
    onButtonClick={[Function]}
  />
  <Button
    buttonEl="0"
    key="0"
    onButtonClick={[Function]}
  />
  <Button
    buttonEl="="
    key="="
    onButtonClick={[Function]}
  />
  <Button
    buttonEl="-"
    key="-"
    onButtonClick={[Function]}
  />
  <Button
    buttonEl="Clear"
    key="Clear"
    onButtonClick={[Function]}
  />
</div>
`;
